service_worker/background.js

```diff
@@ -146,151 +104,114 @@ const hashPassword = async (password) => {
-                if (chrome.runtime.lastError) {
-                    reject(chrome.runtime.lastError);
-                } else {
-                    resolve(data);
-                }
-            });
-        });
-
-        const storedPassword = data.password;
+        const data = await chrome.storage.local.get(['password']);
+        const storedHash = data.password;
 
         const hash = await hashPassword(password);
 
-        if (storedPassword === hash) {
-            
-            try {
-                timeoutDuration = 1 * 60 * 1000
-                startSessionTimeout();
-                chrome.storage.local.set({ loggedIn: true }, () => {
-                    if (chrome.runtime.lastError) {
-                        console.error('Error storing data:', chrome.runtime.lastError);
-                        sendResponse({ success: false });
-                    } else  {
-                        console.log('Kids mode started successfully:');
-                        sendResponse({ success: true });
-    
-                        // Close current windows
-                        chrome.windows.getAll({ populate: true }, (windows) => {
-                            windows.forEach((window) => {
-                                chrome.windows.remove(window.id);
-                            });
-                        });
-    
-                        // Create a new window with Google
-                        chrome.windows.create({
-                            url: '../content/ui/home.html',
-                            type: 'normal'
-                        });
-                    }
-                });
-            }
-            catch (error) {
-                console.error('Error logging in:', error);
-            }}
-            else {
-                // Password mismatch, send error response
-                sendResponse({ success: false, error: '*Invalid credentials' });
-            }
-        }catch(error){
-            console.error('Error logging in:', error);
+        if (!password)
+            return sendResponse({ status: false, error: "Please enter a password" });
+
+        if (storedHash !== hash)
+            return sendResponse({ status: false, error: "Wrong password" });
+
+        try {
+            await injectServiceWorker(checkedToggles)
+            const timeoutDuration = sessionTime * 60 * 60 * 1000; //no. of hrs * 60 min
+            await chrome.storage.local.set({ loggedIn: true, timeLeft: timeoutDuration, sessionTimeout: false })
+
+            sendResponse({ status: true });
+
+            const windows = await chrome.windows.getAll({ populate: true })
+            windows.forEach((window) => {
+                chrome.windows.remove(window.id);
+            });
+
+            await chrome.windows.create({ type: 'normal' });
+        } catch (error) {
+            sendResponse({ status: false, error: "Error logging in, try closing all the windows" });
         }
```

Comment:
Move all of this code to kill old windows and create a new window to a different method, so, that the method can be reused later maybe for a different feature.